Automattic/wp-calypso
# A4A Dev Sites: Add site configuration modal

Related to https://github.com/Automattic/dotcom-forge/issues/8351

## Proposed Changes

![image](https://github.com/user-attachments/assets/daa912e6-928e-403a-aefb-bea5acf49f75)

Opens the site configuration modal when clicking on site creation from `ADD A NEW DEVELOPMENT SITE`
This is for the modal UX only. Calling the provision endpoint will be worked on a separated PR.

## Testing Instructions

- Open the sites page on A4A client.
- Open `Start developing for free` on the header and click to create a new site.
- You should see the site configuration modal.
- The `Allow clients to use the WordPress.com Help Center and hosting features.` checkbox and label should be disabled.

Make a regression test.
Create a new site on A4A by issuing a license and clicking on `Create new site` at the `Needs setup` page.


## Pre-merge Checklist
